presto-hive/src/test/java/com/facebook/presto/hive/TestBackgroundSplitLoader.java

```diff
@@ -0,0 +1,414 @@
+    {
+        BackgroundHiveSplitLoader backgroundHiveSplitLoader = getBackgroundHiveSplitLoader(
+                getFiles(),
+                getPathDomain(returnedPath));
+
+        HiveSplitSource hiveSplitSource = new HiveSplitSource(
+                1,
+                backgroundHiveSplitLoader,
+                executor);
+
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+        assertEquals(1, hiveSplitSource.getOutstandingSplitCount());
+
+        List<ConnectorSplit> connectorSplits = hiveSplitSource.getNextBatch(1).get();
+        assertEquals(1, connectorSplits.size());
+        assertEquals(returnedPath.toString(), ((HiveSplit) connectorSplits.get(0)).getPath());
+    }
+
+    @Test
+    public void testPathFilterOneBucketMatchPartitionedTable()
+            throws Exception
+    {
+        BackgroundHiveSplitLoader backgroundHiveSplitLoader = getBackgroundHiveSplitLoader(
+                getFiles(),
+                getPathDomain(returnedPath),
+                getBuckets(2),
+                getTable(
+                        ImmutableList.of(new Column("partitionColumn", HIVE_INT, Optional.empty())),
+                        getBucketProperty(2)),
+                Optional.empty());
+
+        HiveSplitSource hiveSplitSource = new HiveSplitSource(
+                1,
+                backgroundHiveSplitLoader,
+                executor);
+
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+        assertEquals(1, hiveSplitSource.getOutstandingSplitCount());
+
+        List<ConnectorSplit> connectorSplits = hiveSplitSource.getNextBatch(1).get();
+        assertEquals(1, connectorSplits.size());
+        assertEquals(returnedPath.toString(), ((HiveSplit) connectorSplits.get(0)).getPath());
+    }
+
+    @Test
+    public void testPathFilterBucketedPartitionedTable()
+            throws Exception
+    {
+        Table table = getTable(
+                ImmutableList.of(new Column("partitionColumn", HIVE_INT, Optional.empty())),
+                getBucketProperty(2));
+
+        BackgroundHiveSplitLoader backgroundHiveSplitLoader = getBackgroundHiveSplitLoader(
+                getFiles(),
+                getPathDomain(returnedPath),
+                ImmutableList.of(),
+                table,
+                Optional.of(new HiveBucketHandle(
+                        getRegularColumnHandles(TEST_CONNECTOR_ID, table), 2)));
+
+        HiveSplitSource hiveSplitSource = new HiveSplitSource(
+                1,
+                backgroundHiveSplitLoader,
+                executor);
+
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+        assertEquals(1, hiveSplitSource.getOutstandingSplitCount());
+
+        List<ConnectorSplit> connectorSplits = hiveSplitSource.getNextBatch(1).get();
+        assertEquals(1, connectorSplits.size());
+        assertEquals(returnedPath.toString(), ((HiveSplit) connectorSplits.get(0)).getPath());
+    }
+
+    private List<LocatedFileStatus> getFiles()
```

Comment:
Updated:
1. Moved the necessary functions to be static
2. Removed unnecessary getters, replaced them with ```private static final```
3. For the remaining functions, updated the names are they are more "builders" than getters.